File: lintcode/naive/sort_integers/py/sort_integers3.py
class Solution:
    # @param {int[]} A an integer array
    # @return nothing
    def sortIntegers(self, A):
        # Insertion sort.
        
        for i in xrange(len(A)):
            for j in xrange(i, 0, -1):
                if A[j - 1] <= A[j]:
                    break
                else:
                    A[j], A[j - 1] = A[j - 1], A[j]

File: lintcode/naive/sort_integers/cpp/sort_integers3.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    /**
     * @param A an integer array
     * @return void
     */
    void sortIntegers(std::vector<int>& A) 
    {
        // Insertion sort.

        for (decltype(A.size()) i{ 1 }; i < A.size(); ++i) 
        {
            for (auto j = i; j > 0; --j) {
                if (A[j] < A[j - 1]) {
                    std::swap(A[j], A[j - 1]);
                } else {
                    break;
                }
            }
        }
        
    }
};
